web/libs/editor/src/utils/html.js

```diff
@@ -613,17 +654,51 @@ function sanitizeHtml(html = []) {
     iframe: true,
   };
 
-  return sanitizeHTML(html, {
+  const result = sanitizeHTML(html, {
     allowedTags: false,
     allowedAttributes: false,
     disallowedTagsMode: "discard",
     allowVulnerableTags: true,
     exclusiveFilter(frame) {
-      //...except those in the blacklist
+      // Block scripts unconditionally
+      if (frame.tag === "script") return true;
+
+      // For iframes, only block if NOT from whitelisted domain
+      if (frame.tag === "iframe") {
+        const src = frame.attribs?.src;
+        return !isAllowedIframeSrc(src);
+      }
+
+      // Check other disallowed tags
       return disallowedTags[frame.tag];
     },
     nonTextTags: ["script", "textarea", "option", "noscript"],
     transformTags: {
+      iframe: (tagName, attribs) => {
+        // Only allow specific safe attributes for iframes
+        const allowedIframeAttribs = {};
+        const safeAttrs = [
+          "src",
+          "width",
+          "height",
+          "frameborder",
+          "allowfullscreen",
+          "allow",
+          "title",
+          "referrerpolicy",
+        ];
+
+        safeAttrs.forEach((attr) => {
+          if (attribs[attr]) {
+            allowedIframeAttribs[attr] = attribs[attr];
+          }
+        });
+
+        return {
+          tagName,
+          attribs: allowedIframeAttribs,
+        };
+      },
```

Comment:
only youtube iframes are allowed and they might require some additional attributes, so better to not limit them for now
```suggestion
```